l10n_ch_account_statement_base_import/statement.py

```diff
@@ -71,8 +80,29 @@ def _statement_import(self, cr, uid, ids, prof, parser, file_stream,
         """
         self.file_stream = file_stream
 
-        return super(AccountStatementProfil, self)._statement_import(
-            cr, uid, ids, prof, parser, file_stream, ftype, context)
+        """
+        Calculate hash of the file to determinate if it already exist
+        """
+        hash = hashlib.md5(file_stream).hexdigest()
+        statement_obj = self.pool.get('account.bank.statement')
+
+        for statement_id in statement_obj.search(
+                cr, uid, [('profile_id', '=', prof.id)], context=context):
+            statement = statement_obj.browse(cr, uid, statement_id,
```

Comment:
It would be more efficient to browse all ids at once outside the for loop to reduce number of access to database.